fix: restore /hrbeu/ugs/news.js

Derekmini opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/hrbeu/ugs/news/jwc/jxap
/hrbeu/ugs/news/gztz/all
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

在升级V2的过程中，误将同目录下其他维护者的某个路由删除，通过提交该pr恢复。

Derekmini committed:
fix: restore /heu/ugs/news.js

TonyRL commented on lib/v2/heu/ugs/news.js:
```diff
@@ -0,0 +1,122 @@
+    const author = ctx.params.author || 'gztz';
+    const category = ctx.params.category || 'all';
+    const link = baseUrl + authorMap[author][category] + '/list.htm';
+    const response = await got({
+        method: 'get',
+        url: link,
+        headers: {
+            Referer: baseUrl,
+        },
+    });
```
got request can be simplified
```suggestion
    const response = await got(link, {
        headers: {
            Referer: baseUrl,
        },
    });
```

TonyRL commented on lib/v2/heu/ugs/news.js:
```diff
@@ -0,0 +1,122 @@
+    });
+    const $ = cheerio.load(response.data);
+
+    const urlList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('a', e).attr('href'))
+        .get();
+
+    const titleList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('a', e).text())
+        .get();
+
+    const dateList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('.date', e).text())
+        .get();
```
You can create single list instead of three lists
```suggestion
    const list = $('.wp_article_list_table .border9')
        .toArray()
        .map((e) => {
            e = $(e);
            return {
                title: e.find('a').attr('title'),
                link: new URL(e.find('a').attr('href'), baseUrl).href,
                pubDate: parseDate(e.find('.date').text()),
            };
        });
```

TonyRL commented on lib/v2/heu/ugs/news.js:
```diff
@@ -0,0 +1,122 @@
+        .map((i, e) => $('.date', e).text())
+        .get();
+
+    const out = await Promise.all(
+        urlList.map(async (itemUrl, index) => {
+            itemUrl = url.resolve(baseUrl, itemUrl);
+            if (itemUrl.indexOf('.htm') !== -1) {
+                const cache = await ctx.cache.get(itemUrl);
+                if (cache) {
+                    return Promise.resolve(JSON.parse(cache));
+                }
+                const response = await got.get(itemUrl);
+                const $ = cheerio.load(response.data);
+                const single = {
+                    title: titleList[index],
+                    link: itemUrl,
+                    description: $('.wp_articlecontent')
+                        .html()
+                        .replace(/src="\//g, `src="${url.resolve(baseUrl, '.')}`)
+                        .replace(/href="\//g, `href="${url.resolve(baseUrl, '.')}`)
+                        .trim(),
+                    pubDate: dateList[index],
+                };
+                ctx.cache.set(itemUrl, JSON.stringify(single));
+                return Promise.resolve(single);
+            } else {
+                const single = {
+                    title: titleList[index],
+                    link: itemUrl,
+                    description: '此链接为文件，请点击下载',
+                    pubDate: dateList[index],
+                };
+                return Promise.resolve(single);
+            }
+        })
+    );
```
- `url.resolve()` is deprecated
- Please use `ctx.cache.tryGet()`
```suggestion
    const out = await Promise.all(
        list.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                if (item.link.indexOf('.htm') !== -1) {
                    const response = await got(item.link);
                    const $ = cheerio.load(response.data);

                    item.description = $('.wp_articlecontent')
                        .html()
                        .replace(/src="\//g, `src="${new URL('.', baseUrl).href}`)
                        .replace(/href="\//g, `href="${new URL('.', baseUrl).href}`)
                        .trim();
                } else {
                    item.description = '此链接为文件，请点击下载';
                }
                return item;
            })
        )
    );
```

Derekmini committed:
Update lib/v2/heu/ugs/news.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

Derekmini committed:
Update lib/v2/heu/ugs/news.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

Derekmini committed:
Update lib/v2/heu/ugs/news.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

Derekmini committed:
feat(require): update require

TonyRL committed:
refactor: mise fixes

* rename namespace to hrbeu
- hrbeu.edu.cn is the primary domain name as it's where the contact us mail hosted on

* increase code readability
- use template literals over .concat()
- use .includes() to check the existence of sth

* remove fake publish time

* remove duplicated title in description

TonyRL committed:
fix: fetch article from wechat mp

TonyRL committed:
fix: codeql
